test/storage_init_test.go

```diff
@@ -338,9 +337,139 @@ var _ = Describe(
 				)
 			},
 		)
+
+		Context(
+			"When doing PVC change", func() {
+
+				clusterNamespacedName := getClusterNamespacedName(
+					clusterName, namespace,
+				)
+
+				It(
+					"Should change the pvcUID in initialisedVolumes", func() {
+						zones, err := getZones(ctx, k8sClient)
+						Expect(err).ToNot(HaveOccurred())
+
+						zone1 := zones[0]
+						zone2 := zones[0]
+						racks := []asdbv1.Rack{
+							{ID: 1, Zone: zone1},
+							{ID: 2, Zone: zone2},
+						}
+
+						aeroCluster := createDummyAerospikeCluster(
+							clusterNamespacedName, 2,
+						)
+						rackConf := asdbv1.RackConfig{
+							Racks: racks,
+						}
+						aeroCluster.Spec.RackConfig = rackConf
+
+						err = deployCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						err = validateRackEnabledCluster(
+							k8sClient, ctx, clusterNamespacedName,
+						)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						// validate initialisedVolumes format
+						err = validateInitVolumes(aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						unscheduledPod := clusterName + "-1-0"
+						deletedPVC := "ns-" + unscheduledPod
+
+						oldInitVol := aeroCluster.Status.Pods[unscheduledPod].InitializedVolumes
+
+						By("Unscheduling aerospike pods")
+
+						aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = unschedulableResource()
+
+						err = updateClusterWithTO(k8sClient, ctx, aeroCluster, 1*time.Minute)
+						Expect(err).Should(HaveOccurred())
+
+						By("Cleaning up previous pvc")
+
+						pvcNamespacedName := getPVCNamespacedName(
+							deletedPVC, namespace,
+						)
+
+						err = deletePVC(k8sClient, pvcNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						By("Setting resources to schedule the pods")
+
+						aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster.Spec.PodSpec.AerospikeContainerSpec.Resources = schedulableResource("200Mi")
+
+						err = updateCluster(k8sClient, ctx, aeroCluster)
+						Expect(err).ToNot(HaveOccurred())
+
+						aeroCluster, err = getCluster(k8sClient, ctx, clusterNamespacedName)
+						Expect(err).ToNot(HaveOccurred())
+
+						newInitVol := aeroCluster.Status.Pods[unscheduledPod].InitializedVolumes
+
+						By("Validating initialisedVolumes list")
+
+						err = compareInitialisedVolumes(oldInitVol, newInitVol)
+						Expect(err).ToNot(HaveOccurred())
```

Comment:
This only covers scenario of Pvc UID change.
Shouldn't we add a scenarios/steps where initialisedVolumes contains both pvcname and pvcname@UID and init should cleanup pvcname from the list?